packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts

```diff
@@ -0,0 +1,616 @@
+import { assertNever } from "@fern-api/core-utils";
+import {
+    AliasTypeDeclaration,
+    ContainerType,
+    DeclaredTypeName,
+    EnumTypeDeclaration,
+    ExampleContainer,
+    ExampleObjectProperty,
+    ExampleObjectTypeWithTypeId,
+    ExamplePathParameter,
+    ExamplePrimitive,
+    ExampleRequestBody,
+    ExampleResponse,
+    ExampleSingleUnionType,
+    ExampleSingleUnionTypeProperties,
+    ExampleType,
+    ExampleTypeReference,
+    ExampleTypeReferenceShape,
+    ExampleTypeShape,
+    HttpEndpoint,
+    HttpEndpointExample,
+    HttpHeader,
+    HttpRequestBody,
+    HttpResponse,
+    InlinedRequestBody,
+    IntermediateRepresentation,
+    JsonResponse,
+    JsonResponseBody,
+    JsonResponseBodyWithProperty,
+    Literal,
+    MapType,
+    NameAndWireValue,
+    ObjectProperty,
+    ObjectTypeDeclaration,
+    PathParameter,
+    PrimitiveType,
+    ResponseError,
+    SingleUnionType,
+    SingleUnionTypeProperty,
+    StreamingResponse,
+    TypeDeclaration,
+    TypeId,
+    TypeReference,
+    UndiscriminatedUnionTypeDeclaration,
+    UnionTypeDeclaration
+} from "@fern-api/ir-sdk";
+
+export class ExampleGenerator {
+    ir: IntermediateRepresentation;
+    private types: Map<TypeId, TypeDeclaration>;
+    private MAX_EXAMPLE_DEPTH = 2;
```

Comment:
The autogenerated examples here will also be used in docs if the user doesn't provide them. What we have been following for the OAI is: 
- Request (only contains required properties)
- Response (contains up to depth 2 of auto generated examples, unless the properties are required). 

Some [code](https://github.com/fern-api/fern-definition-registry/blob/d87477ba55d322b9e1cc3577ad5f79d7d45a0440/packages/fdr-sdk/src/converters/db/examples/generateHttpBodyExample.ts#L50-L74) in FDR